command/git.go

```diff
@@ -207,7 +210,17 @@ func (c *gitFetchProject) buildModuleCloneCommand(cloneURI, owner, repo, moduleB
 		}
 		gitCommands = append(gitCommands, cmds...)
 	}
-	gitCommands = append(gitCommands, fmt.Sprintf("git checkout '%s'", ref))
+
+	if requester == evergreen.MergeTestRequester && modulePatch != nil && modulePatch.PatchSet.Patch != "" {
```

Comment:
If the requester is the merge test then we want to check out the PR for this module, assuming a Patch (== PR number) was supplied.

Reply:
should we bundle this into a function?

Reply:
done.